### PR TITLE
[MlN-710] Send confirmation email upon newsletter subscription

### PR DESCRIPTION
**Pull request by gonuguntla**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -124,12 +124,4 @@ def store_user_location!
     # :user is the scope we are authenticating
     store_location_for(:user, originating_location)
   end
-
```
Comment (gonuguntla) [inline app/controllers/application_controller.rb:127]: Below method is not required. So removed it.

File: test/functional/news_letter_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+# TODO: We had some functional tests that tested CRUD (create-update-delete)
```
Comment (daryanypl) [inline test/functional/news_letter_controller_test.rb:5]: did the todo get resolved?  if yes, should we remove the comment and rely on the doc string.  if no, what is the remaining todo functionality?

File: test/functional/news_letter_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# functionality of the teacher sets controller.  They were deleted in
```
Comment (daryanypl) [inline test/functional/news_letter_controller_test.rb:6]: Let's be a little bit more clear.  Some quesitons I have are: 
Who are "they"?  
what are the changes that reflect this new version?

File: test/unit/news_letter_controller_unit_test.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  extend Minitest::Spec::DSL
+  include LogWrapper
+
+  def setup
+    @nl_controller = NewsLetterController.new
+    @mintest_mock1 = MiniTest::Mock.new
+    @mintest_mock2 = MiniTest::Mock.new
+  end
+
+  describe 'Test news-letter email found in google sheets ' do
```
Comment (daryanypl) [inline test/unit/news_letter_controller_unit_test.rb:15]: what are you testing it for?  would you mind making that a bit clearer?

**Review by daryanypl — APPROVED** [review]

File: test/functional/news_letter_controller_test.rb (hunk 3 above)
Comment (gonuguntla) [reply test/functional/news_letter_controller_test.rb:6]: @daryanypl Sorry that is copy paste comments. I will remove it.

File: test/unit/news_letter_controller_unit_test.rb (hunk 4 above)
Comment (gonuguntla) [reply test/unit/news_letter_controller_unit_test.rb]: news-letter email is already in google sheets or not

Comment (gonuguntla) [reply test/unit/news_letter_controller_unit_test.rb]: news-letter email is already in google sheets or not

**Pull request merged**